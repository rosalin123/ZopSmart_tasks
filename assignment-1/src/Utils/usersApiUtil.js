const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchUsers = async () => {
  try {
    let url = `${baseUrl}/users`;
    let users = await fetch(url, { method: 'GET' });
    let usersJson = await users.json();
    return usersJson;
  } catch (err) {
    console.log('Error fetching users details', err.stack);
  }
};

export const getUser = async (id) => {

    let url = `${baseUrl}/users/${id}`;
    let user = await fetch(url, { method: 'GET' });
    let userJson = await user.json();
    return userJson;

};
