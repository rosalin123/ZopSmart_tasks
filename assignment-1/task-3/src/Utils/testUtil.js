import React from 'react';

export const findbyTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};
