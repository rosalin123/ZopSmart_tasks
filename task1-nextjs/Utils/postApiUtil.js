import axios from 'axios';
import fetch from 'isomorphic-unfetch';
const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchPosts = async () => {
  let url = `${baseUrl}/posts`;
  let posts = await axios.get(url);
  let postsJson = await posts.json();
  return postsJson;
};

export const getPost = async (id) => {
  let url = `${baseUrl}/posts/${id}`;
  let post = fetch(url);
  let postJson = await post.json();
  return postJson;
};

export const getPostComments = async (id) => {
  let url = `${baseUrl}/posts/${id}/comments`;
  let comments = await axios.get(url);
  let commentsJson = await comments.json();
  return commentsJson;
};

export const getUserPosts = async (id) => {
  let url = `${baseUrl}/posts?userId=${id}`;
  let userPosts = await fetch(url);
  let userPostsJson = await userPosts.json();
  return userPostsJson;
};
