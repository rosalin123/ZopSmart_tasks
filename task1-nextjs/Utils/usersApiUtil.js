import fetch from 'isomorphic-unfetch';
const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchUsers = async () => {
  try {
    let url = `${baseUrl}/users`;
    let users = await fetch(url);
    let usersJson = await users.json();
    return usersJson;
  } catch (err) {
    console.log('Error fetching users details', err.stack);
  }
};

export const getUser = async (id) => {
  let url = `${baseUrl}/users/${id}`;
  let user = await fetch(url);
  let userJson = await user.json();
  return userJson;
};
