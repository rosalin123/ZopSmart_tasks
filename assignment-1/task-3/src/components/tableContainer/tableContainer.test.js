import React from 'react';
import { shallow, mount, render } from 'enzyme';
import TableContainer from './index';
import { findbyTestAttr } from '../../Utils/testUtil';

const setUp = () => {
  const component = shallow(<TableContainer />).dive();
  return component;
};

describe('TableContainer component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it('TableContainer should render without throuwing any error', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render header', () => {
    let wrapper = findbyTestAttr(component, 'header');
    expect(wrapper.length).toBe(1);
  });

  it('should render form', () => {
    let wrapper = findbyTestAttr(component, 'form');
    expect(wrapper.length).toBe(1);
  });

  it('should render input fields', () => {
    let wrapper = findbyTestAttr(component, 'input');
    expect(wrapper.length).toBe(2);
  });

  it('should render submit button', () => {
    let wrapper = findbyTestAttr(component, 'submit');
    expect(wrapper.length).toBe(1);
  });

  it('check submit', () => {
    const formEventMocked = { preventDefault: jest.fn() };

    findbyTestAttr(component, 'form').simulate('submit', formEventMocked);
    expect(formEventMocked.preventDefault).toBeCalledTimes(1);
    expect(component.state('showTable')).toBe(true);
  });

  it('should call onChange prop', () => {
    const event = {
      target: { value: 2 },
    };
    findbyTestAttr(component, 'columnInput').simulate('change', event);
    findbyTestAttr(component, 'rowsInput').simulate('change', event);
    expect(component.state('columns')).toBe(2);
    expect(component.state('rows')).toBe(2);
  });
});
