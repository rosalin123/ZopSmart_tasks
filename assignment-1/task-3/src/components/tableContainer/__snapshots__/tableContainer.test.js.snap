// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableContainer component TableContainer should render without throuwing any error 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="TableContainer-contentStyles-1"
  container={true}
  direction="column"
>
  <WithStyles(ForwardRef(Grid))
    className="TableContainer-headerStyles-3"
    data-test="header"
    item={true}
  >
    <Header />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="TableContainer-formStyles-2"
    item={true}
  >
     
    <form
      data-test="form"
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        direction="column"
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <h3>
            Enter rows and columns to create a table
          </h3>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Grid))
            className="TableContainer-paddingStyles-5"
            container={true}
            direction="column"
            spacing={1}
          >
             
            <WithStyles(ForwardRef(Grid))
              data-test="input"
              item={true}
            >
              <WithStyles(ForwardRef(Grid))
                container={true}
                justify="space-between"
                spacing={2}
              >
                 
                <WithStyles(ForwardRef(Grid))
                  item={true}
                >
                   
                  <WithStyles(ForwardRef(Typography))
                    className="TableContainer-typograpgyStyles-7"
                    color="textSecondary"
                  >
                    Rows:
                  </WithStyles(ForwardRef(Typography))>
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                >
                   
                  <input
                    className="TableContainer-inputFieldStyles-4"
                    data-test="rowsInput"
                    onChange={[Function]}
                    type="number"
                  />
                </WithStyles(ForwardRef(Grid))>
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              data-test="input"
              item={true}
            >
              <WithStyles(ForwardRef(Grid))
                container={true}
                justify="space-between"
              >
                 
                <WithStyles(ForwardRef(Grid))
                  item={true}
                >
                   
                  <WithStyles(ForwardRef(Typography))
                    className="TableContainer-typograpgyStyles-7"
                    color="textSecondary"
                  >
                    Columns:
                  </WithStyles(ForwardRef(Typography))>
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                >
                  <input
                    className="TableContainer-inputFieldStyles-4"
                    data-test="columnInput"
                    onChange={[Function]}
                    type="number"
                  />
                </WithStyles(ForwardRef(Grid))>
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
           
          <WithStyles(ForwardRef(Button))
            className="TableContainer-submitButtonStyles-6"
            color="primary"
            data-test="submit"
            type="submit"
            variant="contained"
          >
            Submit
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </form>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="TableContainer-tableStyles-8"
    item={true}
  >
     
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
